src/generators/legacy-json/index.mjs

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+import { writeFile } from 'node:fs/promises';
+import { join } from 'node:path';
+import { groupNodesByModule } from '../../utils/generators.mjs';
+import buildSection from './utils/buildSection.mjs';
+
+/**
+ * This generator is responsible for generating the legacy JSON files for the
+ *  legacy API docs for retro-compatibility. It is to be replaced while we work
+ *  on the new schema for this file.
+ *
+ * This is a top-level generator, intaking the raw AST tree of the api docs.
+ * It generates JSON files to the specified output directory given by the
+ *  config.
+ *
+ * @typedef {Array<ApiDocMetadataEntry>} Input
+ *
+ * @type {import('../types.d.ts').GeneratorMetadata<Input, import('./types.d.ts').Section[]>}
+ */
+export default {
+  name: 'legacy-json',
+
+  version: '1.0.0',
+
+  description: 'Generates the legacy version of the JSON API docs.',
+
+  dependsOn: 'ast',
+
+  async generate(input, { output }) {
+    // This array holds all the generated values for each module
+    const generatedValues = [];
+
+    const groupedModules = groupNodesByModule(input);
+
+    // Gets the first nodes of each module, which is considered the "head"
+    const headNodes = input.filter(node => node.heading.depth === 1);
+
+    /**
+     * @param {ApiDocMetadataEntry} head
+     * @returns {import('./types.d.ts').ModuleSection}
+     */
+    const processModuleNodes = head => {
+      const nodes = groupedModules.get(head.api);
+
+      const section = buildSection(head, nodes);
+      generatedValues.push(section);
+
+      return section;
+    };
+
+    for (const node of headNodes) {
+      // Get the json for the node's section
+      const section = processModuleNodes(node);
+
+      // Write it to the output file
+      await writeFile(
+        join(output, `${node.api}.json`),
+        JSON.stringify(section),
+        'utf8'
+      );
+    }
```

Comment:
```suggestion
    await Promise.all(headNodes.map(async node => {
      // Get the json for the node's section
      const section = processModuleNodes(node);
    
      // Write it to the output file
      await writeFile(
        join(output, `${node.api}.json`),
        JSON.stringify(section),
        'utf8'
      );
    }));
```

I don't know why, but I discovered that making promise.all was faster than await at each iteration.
OFC we can catch if there are an error

Reply:
It's faster since it makes all the nodes be processed in parallel. Nothing's reliant on the nodes being processed in sequential order so we can definitely do this, nice catch